drivers/mysql/mysql.go

```diff
@@ -0,0 +1,395 @@
+// Initial code generated by generator.
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/go-morph/morph/drivers"
+	"github.com/go-morph/morph/models"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+var driverName = "mysql"
+var defaultMigrationMaxSize = 10 * 1 << 20 // 10 MB
+var defaultConfig = &Config{
+	MigrationsTable:        "schema_migrations",
+	StatementTimeoutInSecs: 5,
+	MigrationMaxSize:       defaultMigrationMaxSize,
+}
+
+// add here any custom driver configuration
+var configParams = []string{
+	"x-migration-max-size",
+	"x-migrations-table",
+	"x-statement-timeout",
+}
+
+func init() {
+	drivers.Register("mysql", &mysql{})
+}
+
+type Config struct {
+	MigrationsTable        string
+	StatementTimeoutInSecs int
+	MigrationMaxSize       int
+	databaseName           string
+}
+
+type mysql struct {
+	conn   *sql.Conn
+	db     *sql.DB
+	config *Config
+}
+
+func WithInstance(dbInstance *sql.DB, config *Config) (drivers.Driver, error) {
+	driverConfig := mergeConfigs(config, defaultConfig)
+
+	conn, err := dbInstance.Conn(context.Background())
+	if err != nil {
+		return nil, &drivers.DatabaseError{Driver: driverName, Command: "grabbing_connection", OrigErr: err, Message: "failed to grab connection to the database"}
+	}
+
+	if driverConfig.databaseName, err = currentDatabaseNameFromDB(conn, driverConfig); err != nil {
+		return nil, err
+	}
+
+	return &mysql{config: config}, nil
+}
+
+func (driver *mysql) Open(connURL string) (drivers.Driver, error) {
+	customParams, err := drivers.ExtractCustomParams(connURL, configParams)
+	if err != nil {
+		return nil, &drivers.AppError{Driver: driverName, OrigErr: err, Message: "failed to parse custom parameters from url"}
+	}
+
+	sanitizedConnURL, err := drivers.RemoveParamsFromURL(connURL, configParams)
+	if err != nil {
+		return nil, &drivers.AppError{Driver: driverName, OrigErr: err, Message: "failed to sanitize url from custom parameters"}
+	}
+
+	driverConfig, err := mergeConfigWithParams(customParams, defaultConfig)
+	if err != nil {
+		return nil, &drivers.AppError{Driver: driverName, OrigErr: err, Message: "failed to merge custom params to driver config"}
+	}
+
+	db, err := sql.Open(driverName, sanitizedConnURL)
+	if err != nil {
+		return nil, &drivers.DatabaseError{Driver: driverName, Command: "opening_connection", OrigErr: err, Message: "failed to open connection with the database"}
+	}
+
+	conn, err := db.Conn(context.Background())
```

Comment:
What about adding this as an argument? I mean it could be useful for the caller. In fact, could `driver` share this context for every other method? e.g. ping, lock etc.

Reply:
I think we should keep this as it is for now to maintain compatibility with migrate's interface, but I'm happy to explore this idea :)